New README

tlienart opened the pull request:
Preview: https://github.com/tlienart/MLJ.jl/blob/master/README.md

ablaom commented on README.md:
````diff
@@ -50,6 +51,41 @@ MLJ is released under the MIT licensed and sponsored by the [Alan Turing Institu
+```
+
+For a list of models and their packages see the [table below](#models-available), we recommend you start with models marked as coming from _mature_ packages such as _DecisionTree_, _ScikitLearn_ or _XGBoost_.
+
````
Replace with
"For a list of models and their packages  set the [table below](#models-available) or run `using MLJ; models()`. We recommend ..."

ablaom commented on README.md:
```diff
@@ -173,24 +195,11 @@ The table below indicates the models that are accessible at present along with a
+
+*Past collaborators*: D. Aluthge, E. Barp, G. Bohner, M. K. Borregaard, V. Churavy, H. Devereux, M. Giordano, M. Innes, F. Kiraly, M. Nook, Z. Nugent, P. Oleśkiewicz, A. Shridar, Y. Simillides, A. Sengupta, A. Stechemesser.
 
 #### License
```
I have given this some thought, and I think it is important to acknowledge Sebastian and Franz in the design. Can we please add:

*Core design*: A. Blaom, F. Kiraly, S. Vollmer

ablaom commented:
Thanks again for this @tlienart 

The links at the top for Cheatsheet and Citing MLJ need fixing.

ablaom commented on README.md:
Link to wrong repo

ablaom commented on README.md:
Why is KNNRegressor here? Is this a reference to the obsolete version? I think we can just drop this.

ablaom commented on README.md:
Finicky request: replace "[note]:" with "*Note*:"

tlienart replied:
yes so in proper markdown the `note` should have appeared as a link, it seems Github flavoured one doesn't resolve it... fixed as per your comment

ablaom commented:
Please remove "GaussianProcesses" from the package list. The implementation there is wrong and the package has been removed some time ago from the registry.  

Also, I would demote "NaiveBayes" maturity to "low". It is old and not well-maintained, has a suspected bug, and our implementation does not support sparsity. (The package itself does, but in a substandard way using dictionaries.)

ablaom commented:
Also, I though in the "help wanted" section we could add something about improving sparse data support and support for natural language processing. What do think?

If you are happy with my comments, no need for me to revisit. Merge when ready.

tlienart commented:
* cheatsheet, citing link fixed
* added `models()`
* added `core design`
* fixed link to mljmodels, sklearn
* removed knnregressor from builtins
* added a note about sparsity  and NLP in the help section.

Will double check then merge
